src/legacy/ui/public/courier/fetch/call_client.js

```diff
@@ -17,187 +17,81 @@
+  if (response.timed_out) {
+    toastNotifications.addWarning({
+      title: i18n.translate('common.ui.courier.fetch.requestTimedOutNotificationMessage', {
+        defaultMessage: 'Data might be incomplete because your request timed out',
+      }),
     });
+  }
 
-    const searchStrategiesWithRequests = assignSearchRequestsToSearchStrategies(requestsToFetch);
-
-    // We're going to create a new async context here, so that the logic within it can execute
-    // asynchronously after we've returned a reference to defer.promise.
-    Promise.resolve().then(async () => {
-      // Execute each request using its search strategy.
-      for (let i = 0; i < searchStrategiesWithRequests.length; i++) {
-        const searchStrategyWithSearchRequests = searchStrategiesWithRequests[i];
-        const { searchStrategy, searchRequests } = searchStrategyWithSearchRequests;
-        const {
-          searching,
-          abort,
-          failedSearchRequests,
-        } = await searchStrategy.search({ searchRequests, es, Promise, serializeFetchParams, config, sessionId, esShardTimeout });
-
-        // Collect searchRequests which have successfully been sent.
-        searchRequests.forEach(searchRequest => {
-          if (failedSearchRequests.includes(searchRequest)) {
-            return;
-          }
-
-          activeSearchRequests.push(searchRequest);
-        });
-
-        abortableSearches.push({
-          searching,
-          abort,
-          requestsCount: searchRequests.length,
-        });
-      }
-
-      try {
-        // The request was aborted while we were doing the above logic.
-        if (areAllSearchRequestsAborted) {
-          return;
-        }
-
-        const segregatedResponses = await Promise.all(abortableSearches.map(async ({ searching, requestsCount }) => {
-          return searching.catch((e) => {
-            // Duplicate errors so that they correspond to the original requests.
-            return new Array(requestsCount).fill({ error: e });
-          });
-        }));
-
-        // Assigning searchRequests to strategies means that the responses come back in a different
-        // order than the original searchRequests. So we'll put them back in order so that we can
-        // use the order to associate each response with the original request.
-        const responsesInOriginalRequestOrder = new Array(searchRequests.length);
-        segregatedResponses.forEach((responses, strategyIndex) => {
-          responses.forEach((response, responseIndex) => {
-            const searchRequest = searchStrategiesWithRequests[strategyIndex].searchRequests[responseIndex];
-            const requestIndex = searchRequests.indexOf(searchRequest);
-            responsesInOriginalRequestOrder[requestIndex] = response;
-          });
-        });
-
-        await respondToSearchRequests(responsesInOriginalRequestOrder);
-      } catch(error) {
-        if (errorAllowExplicitIndex.test(error)) {
```

Comment:
Right now if you set `rest.action.multi.allow_explicit_index` to `false` in your elasticsearch.yml, a lot of things are broken (including the saved objects client and add sample data). I don't see any reason to explicitly handle this here because the apps also rely on the saved objects client, which doesn't handle this. Maybe we should open an issue to better handle this scenario from a higher level.